examples/django/app/views.py

```diff
@@ -0,0 +1,39 @@
+import json
+from datetime import *
+import simplejson
+from django.http import HttpResponse, JsonResponse, HttpResponseNotFound
+from rest_framework import status
+from rest_framework.utils import json
+
+
+# Create your views here.
+city_list = ["pune", "austin", "santabarbara"]
+
+
+# API for Hello Application
+def hello(request):
+    return HttpResponse("Hello, Welcome to the Weather App!")
+
+
+# API for current server date
+def getdate(request):
+    current_datetime = datetime.now()
+    html = "Current Date and Time is: %s" % current_datetime
+    return HttpResponse(html, {"status": "success"}, status=status.HTTP_200_OK)
+
+
+# API for location details
+def get_location_details(request):
+    return HttpResponse(simplejson.dumps(city_list), content_type='application/json', status=status.HTTP_200_OK)
+
+
+# API for weather details for a location
+def get_weather_details(request, location_name):
+    file_name = location_name + ".json"
+    if location_name in city_list:
+        try:
+            with open('static/' + file_name) as f:
+                data = json.load(f)
+        except IOError:
+            return HttpResponseNotFound("No Response")
+    return JsonResponse(data, status=status.HTTP_200_OK, safe=False)
```

Comment:
There is a bug here, but let's keep it and explain why it's a bug.
```suggestion
    # This is an intentional bug that occurs when a `location_name` is passed but is not in `city_list`.
    return JsonResponse(data, status=status.HTTP_200_OK, safe=False)
```

The error is:
```
UnboundLocalError:  local variable 'data' referenced before assignment
```

I found this by running the example app and then using this curl command:

```
curl -I "http://localhost:3000/weather/washington"
```

In Airbrake, I received the error explaining the bug.